composer/models/deeplabv3/model.py

```diff
@@ -94,7 +97,12 @@ def deeplabv3(num_classes: int,
              https://download.openmmlab.com/mmcv/dist/{cu_version}/{torch_version}/index.html where {cu_version} and
              {torch_version} refer to your CUDA and PyTorch versions, respectively. To install mmsegmentation, please
              run pip install mmsegmentation==0.22.0 on command-line.""")) from e
-    norm_type = 'SyncBN' if sync_bn else 'BN'
+
+    world_size = dist.get_world_size()
+    if sync_bn and world_size == 1:
+        warnings.warn('sync_bn was true, but only one process is present for training. sync_bn will be ignored.')
+
+    norm_type = 'SyncBN' if sync_bn and world_size > 1 else 'BN'
```

Comment:
if the `norm_cfg` is already set to `SyncBN`, do we still need to run the `convert_sync_batchnorm` call below?

Reply:
`norm_cfg` is only for the head of network i.e. DeepLabv3+. The `convert_sync_batchnorm` is primarily for the backbone, i.e. ResNet. It may be the case that we only need `convert_sync_batchnorm`, but since mmseg has a lot of their own bespoke layers, I'm not sure `convert_sync_batchnorm` detects the custom layers. I can test this if needed.

Reply:
ok thanks. not neccessary, better to be safe I suppose!